lib/shoryuken/manager.rb

```diff
@@ -15,6 +15,8 @@ def initialize(fetcher, polling_strategy)
       @fetcher = fetcher
       @polling_strategy = polling_strategy
 
+      @heartbeat = Concurrent::TimerTask.new(run_now: true, execution_interval: 1, timeout_interval: 60) { dispatch }
```

Comment:
@mariokostelac ^ is that what you were saying about only fetching on a second internal?

If so, I can revert to my previous solution, which tries to fetcher whenever a processor is available.

```ruby
def processor_done(queue)
  logger.debug { "Process done for '#{queue}'" }

  @ready.increment

  dispatch
end

def dispatch
  return if @done.true?

  # ...

  dispatch_later
end

def dispatch_later
  return unless @dispatch_later.make_true

  after(1) do
    @dispatch_later.make_false
    dispatch
  end
end
```

But this may cause requests to SQS with a batch size of 1, in case only one processor gets available.